unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Loader/Loader.cs

```diff
@@ -0,0 +1,202 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Helpers;
+using UnityEngine;
+
+namespace AvatarSystem
+{
+    public class Loader : ILoader
+    {
+        public GameObject bodyshapeContainer => bodyshapeLoader?.rendereable?.container;
+        public SkinnedMeshRenderer combinedRenderer { get; private set; }
+        public Renderer eyesRenderer { get; private set; }
+        public Renderer eyebrowsRenderer { get; private set; }
+        public Renderer mouthRenderer { get; private set; }
+        public ILoader.Status status { get; private set; } = ILoader.Status.Idle;
+
+        private readonly IWearableLoaderFactory wearableLoaderFactory;
+        private readonly GameObject container;
+
+        private IBodyshapeLoader bodyshapeLoader;
+        private readonly Dictionary<string, IWearableLoader> loaders = new Dictionary<string, IWearableLoader>();
+        private readonly AvatarMeshCombinerHelper avatarMeshCombiner;
+
+        public Loader(IWearableLoaderFactory wearableLoaderFactory, GameObject container)
+        {
+            this.wearableLoaderFactory = wearableLoaderFactory;
+            this.container = container;
+
+            avatarMeshCombiner = new AvatarMeshCombinerHelper();
+            avatarMeshCombiner.prepareMeshForGpuSkinning = true;
+            avatarMeshCombiner.uploadMeshToGpu = true;
+        }
+
+        public async UniTask Load(WearableItem bodyshape, WearableItem eyes, WearableItem eyebrows, WearableItem mouth, List<WearableItem> wearables, AvatarSettings settings, CancellationToken ct = default)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            List<IWearableLoader> toCleanUp = new List<IWearableLoader>();
+
+            void DisposeCleanUpLoaders()
+            {
+                for (int i = 0; i < toCleanUp.Count; i++)
+                {
+                    if (toCleanUp[i] == null)
+                        continue;
+                    toCleanUp[i].Dispose();
+                }
+            }
+
+            try
+            {
+                status = ILoader.Status.Loading;
+
+                if (bodyshapeLoader == null || bodyshapeLoader.wearable.id != bodyshape.id)
+                {
+                    toCleanUp.Add(bodyshapeLoader);
+                    bodyshapeLoader = wearableLoaderFactory.GetBodyshapeLoader(bodyshape, eyes, eyebrows, mouth);
+                }
+
+                await bodyshapeLoader.Load(container, settings, ct);
+
+                if (bodyshapeLoader.status == IWearableLoader.Status.Failed)
+                {
+                    status = ILoader.Status.Failed_Mayor;
+                    return;
+                }
+
+                // Mark for cleanUp unneeded loaders
+                List<string> unnededCategories = new List<string>();
+                foreach ((string category, IWearableLoader existentLoader) in loaders)
```

Comment:
you could extract into a method the solving of loaders for the wearables: `SolveLoadersForWearables` or `UpdateLoadersForWearables`